src/CodeXMLParse.cpp

```diff
@@ -46,17 +45,67 @@ void AnnotateOpref(ANNOTATOR_PARAMS)
 	annotation.type = R_CODE_ANNOTATION_TYPE_OFFSET;
 	annotation.offset.offset = op->getAddr().getOffset();
 }
-void AnnotateFunctionName(ANNOTATOR_PARAMS){
-	auto func_name = node.child_value();
+void AnnotateFunctionName(ANNOTATOR_PARAMS)
+{
+	const char *func_name = node.child_value();
 	if(!func_name)
 		return;
 	RCodeAnnotation annotation = {};
 	annotation.type = R_CODE_ANNOTATION_TYPE_FUNCTION_NAME;
-	annotation.function_name.name = func_name;
+	pugi::xml_attribute attr = node.attribute("opref");
+	if(attr.empty())
+	{
+		if(ctx->func->getName() == func_name)
+		{
+			annotation.function_name.name = strdup(ctx->func->getName().c_str());
+			annotation.function_name.offset = ctx->func->getAddress().getOffset();
+			// Code below makes an offset annotation for the function name(for the currently decompiled function)
+			RCodeAnnotation offsetAnnotation = {};
+			offsetAnnotation.type = R_CODE_ANNOTATION_TYPE_OFFSET;
+			offsetAnnotation.offset.offset = annotation.function_name.offset;
+			out->push_back(offsetAnnotation);
+		}
+		else
+		{
+			annotation.function_name.name = strdup("INVALID: Doesn't look like a function.");
+			annotation.function_name.offset = UINT64_MAX;
+		}
+		out->push_back(annotation);
+		return;
+	}
+	unsigned long long opref = attr.as_ullong(ULLONG_MAX);
+	if(opref == ULLONG_MAX)
+	{
+		annotation.function_name.name = strdup("INVALID: Doesn't look like a function.");
+		annotation.function_name.offset = UINT64_MAX;
+		out->push_back(annotation);
+		return;
+	}
+	auto opit = ctx->ops.find((uintm)opref);
+	if(opit == ctx->ops.end())
+	{
+		annotation.function_name.name = strdup("INVALID: Doesn't look like a function.");
+		annotation.function_name.offset = UINT64_MAX;
+		out->push_back(annotation);
+		return;	
+	}
+	PcodeOp *op = opit->second;
+	FuncCallSpecs *call_func_spec = ctx->func->getCallSpecs(const_cast<PcodeOp *>(op));
```

Comment:
Why did you decide to do a const cast here?

Reply:
`FuncCallSpecs *getCallSpecs(const PcodeOp *op) const;`
Because it accepts a constant PcodeOp as the argument. Now I think I should remove it.

Reply:
Code should almost never contain contain const_casts. Valid uses cases for it are very rare. In this case you are  not even changing the type using the cast. You might want to read a bit more about how const works.

Reply:
Thanks, I understood why it's useless in this case.

Reply:
Removed const cast.